app/uk/gov/hmrc/trusts/services/BeneficiaryTransformationService.scala

```diff
@@ -17,19 +17,47 @@
 package uk.gov.hmrc.trusts.services
 
 import javax.inject.Inject
+import play.api.libs.json.{JsObject, __}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.trusts.exceptions.InternalServerErrorException
+import uk.gov.hmrc.trusts.models.get_trust_or_estate.get_trust.TrustProcessedResponse
 import uk.gov.hmrc.trusts.models.variation.UnidentifiedType
+import uk.gov.hmrc.trusts.models.{RemoveBeneficiary, Success}
 import uk.gov.hmrc.trusts.transformers._
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
-class BeneficiaryTransformationService @Inject()(transformationService: TransformationService) {
+class BeneficiaryTransformationService @Inject()(transformationService: TransformationService)(implicit ec:ExecutionContext) {
+  def removeBeneficiary(utr: String, internalId: String, removeBeneficiary: RemoveBeneficiary)(implicit hc: HeaderCarrier) : Future[Success.type] = {
+    transformationService.getTransformedData(utr, internalId).flatMap {
```

Comment:
@JoeSimmonds Could we extract the core of this 'picking' into a separate function named 'getBeneficiaryJson(type, index)' which returns a the future we can map over?

Reply:
Not sure if this is what you meant